cmd/kube-apiserver/app/server.go

```diff
@@ -687,6 +689,11 @@ func Complete(s *options.ServerRunOptions) (completedServerRunOptions, error) {
 				return options, fmt.Errorf("the serviceaccount max expiration must be between 1 hour to 2^32 seconds")
 			}
 		}
+		// When service-account-extend-token-expiration is enabled during transition time, max expiration must be set
+		// long enough for fixing clients.
+		if s.Authentication.ServiceAccounts.ExtendExpiration && s.Authentication.ServiceAccounts.MaxExpiration < 365*24*time.Hour {
```

Comment:
move this check inside the `if s.Authentication.ServiceAccounts.MaxExpiration != 0 {` block

Reply:
Done